packit/source_git.py

```diff
@@ -0,0 +1,364 @@
+                'by running `rpmbuild -bp` but we cannot find "packitpatch" command on PATH: '
+                "please install packit as an RPM."
+            )
+        logger.info(
+            f"expanding %prep section in {self.dist_git.local_project.working_dir}"
+        )
+        user_rpmmacros_path = Path("~/.rpmmacros").expanduser()
+        user_rpmmacros_path_bkp = Path(f"{user_rpmmacros_path}.bkp")
+        restore = False
+
+        if user_rpmmacros_path.is_file():
+            logger.debug(
+                f"{user_rpmmacros_path} exists, backing it up and replacing with ours"
+            )
+            shutil.move(user_rpmmacros_path, user_rpmmacros_path_bkp)
+            restore = True
+        user_rpmmacros_path.write_text(RPM_MACROS_FOR_PREP)
+
+        try:
+            rpmbuild_args = [
+                "rpmbuild",
+                "--nodeps",
+                "--define",
+                f"_topdir {str(self.dist_git.local_project.working_dir)}",
+                "-bp",
+            ]
+            rpmbuild_args += [
+                "--define",
+                f"_specdir {str(self.dist_git.absolute_specfile_dir)}",
+            ]
+            rpmbuild_args += [
+                "--define",
+                f"_sourcedir {str(self.dist_git.absolute_source_dir)}",
+            ]
+            if logger.level <= logging.DEBUG:  # -vv can be super-duper verbose
+                rpmbuild_args.append("-v")
+            rpmbuild_args.append(str(self.dist_git.absolute_specfile_path))
+
+            run_command(
+                rpmbuild_args,
+                cwd=self.dist_git.local_project.working_dir,
+                print_live=True,
+            )
+        finally:
+            user_rpmmacros_path.unlink()
+            if restore:
+                logger.debug(
+                    f"restoring {user_rpmmacros_path_bkp} as {user_rpmmacros_path}"
+                )
+                shutil.move(user_rpmmacros_path_bkp, user_rpmmacros_path)
+
+    def _put_downstream_sources(self):
+        root_downstream_dir = self.dist_git.get_root_downstream_dir_for_source_git(
+            self.local_project.working_dir
+        )
+        os.makedirs(root_downstream_dir, exist_ok=True)
+
+        shutil.copy2(self.dist_git.absolute_specfile_path, root_downstream_dir)
+
+        logger.info(
+            f"Copy all sources from {self.dist_git.absolute_source_dir} to {root_downstream_dir}."
+        )
+
+        # we may not want to copy the primary archive - it's worth a debate
+        for source in self.dist_git.specfile.get_sources():
+            source_dest = root_downstream_dir / Path(source).name
+            logger.debug(f"copying {source} to {source_dest}")
+            shutil.copy2(source, source_dest)
+
+        self.local_project.stage(self.dist_git.source_git_downstream_suffix)
+        self.local_project.commit(message="add downstream distribution sources")
+
+    def _add_packit_xml(self):
```

Comment:
xml?

Reply:
lol, I actually left it there - will fix